### PR TITLE
Feature/views

### PR DESCRIPTION
**Pull request by leolpaz**
### Ruby/Blog-App

Added on this pull request:

- User, Posts
  - Views
  - Methods to pass information to the views

### CODE REVIEW

File: app/views/users/index.html.erb [hunk 1]
```diff
@@ -1,2 +1,10 @@
...
+<h1>User list</h1>
+<ul>
+  <% @users.each do |user|%>
+    <li>
+      <img src="<% user.photo %>" alt="user photo">
+      <h2><%= link_to user.name, user %></h2>
+      <span>Number of posts: <% user.posts_counter %></span>
+    </li>
+  <% end %>
+</ul>
```
Comment (Lameck1) [inline app/views/users/index.html.erb]: - [OPTIONAL] Please, let's follow the provided wireframe to achieve the desired layout of this section. This is how the elements in this section should be aligned:

  image
  
  This is how your app displays this section:
  
  image
  
  As evident above, the image, the name of the user should be displayed in a horizontal alignment.
  
  The problem is using `h2` which is a block-level element and thus every other element is forced to be in its own line.

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -1,2 +1,10 @@
-<h1>Users#index</h1>
-<p>This is where the users list will be shown</p>
+<h1>User list</h1>
+<ul>
+  <% @users.each do |user|%>
+    <li>
+      <img src="<% user.photo %>" alt="user photo">
```
Comment (Lameck1) [inline app/views/users/index.html.erb]: - [x] There is a bug here that prevents us from displaying the user's photo. You forgot the assignment operator to tell ERB that this is a value you intend to show on the screen. Please refactor as below:

  ```ERB
  <img src="<%= user.photo %>" alt="user photo">
  ```

File: app/views/users/index.html.erb [hunk 3]
```diff
@@ -1,2 +1,10 @@
...
+      <h2><%= link_to user.name, user %></h2>
+      <span>Number of posts: <% user.posts_counter %></span>
```
Comment (Lameck1) [inline app/views/users/index.html.erb]: - [x] Number of user's posts are also not being displayed. Same issue as above. Please refactor as follows:
  
  ```ERB
  <span>Number of posts: <%= user.posts_counter %></span>
  ```

File: app/views/users/_user.html.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<div>
+  <img src="<%= @user.photo %>" alt="user picture">
+  <div>
+    <h2><%= @user.name %></h2>
+  <span>Number of posts: <% @user.posts_counter %></span>
```
Comment (Lameck1) [inline app/views/users/_user.html.erb]: - [x] Let's also refactor this to help us show post_counter on the page:
  
  ```ERB
  <span>Number of posts: <%= @user.posts_counter %></span>
  ```

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -1,2 +1,10 @@
...
+<h1>User details</h1>
+<%= render "user" %>
+<ul>
+  <li>
+    <h2>Bio</h2>
+    <p><%= @user.bio %></p>
+  </li>
+  <%= render "posts" %>
+</ul>
+<button><%= link_to "See all posts", [@user, :posts] %></button>
```
Comment (Lameck1) [inline app/views/users/show.html.erb:10]: - [OPTIONAL] Let's try and match the wireframe for the User details section too. Below is how the page should at least look like in terms of the arrangement of different parts of the user's details:
  image

File: app/views/posts/index.html.erb [hunk 6]
```diff
@@ -1,2 +1,13 @@
-<h1>Posts#index</h1>
-<p>Here will be a list of user posts</p>
+<h1>Post list</h1>
+<%= render "users/user" %>
+<% @user.posts.each do |post| %>
+  <%= link_to [post.user, post] do %></h2>
+    <div>
+      <h3><%= post.title %></h3>
+      <p><%= post.text %></p>
+    </div>
```
Comment (Lameck1) [inline app/views/posts/index.html.erb]: - [ ] Within this `div`, we are expected to display the number of comments and the number of likes as in the wireframe. Please include these details. See the wireframe for this particular part below:
  
  image

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi, @leolpaz, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/users/index.html.erb (hunk 1 above)
Comment (leolpaz) [reply app/views/users/index.html.erb]: Thanks for the review, I'll do the entire design on Friday, as I'm struggling a bit with Rails and am focusing all of my current time on it

File: app/views/users/show.html.erb (hunk 5 above)
Comment (leolpaz) [reply app/views/users/show.html.erb:10]: Same as above

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**